File: id-alejandrogreenwick/Refactor/refactor.py
# refactor.py (v1 - intentionally messy)
import sys, random

nums = [] 

def readf(p):
    f = open(p)  
    for ln in f:
        if ln.strip() == "":
            continue
        try:
            x = int(ln.strip())
        except:
            x = 0  
        nums.append(x)
    f.close()

def stats():
    mn = 9999999
    mx = -9999999
    s = 0
    for x in nums:
        if x < mn: mn = x
        if x > mx: mx = x
        s += x
    return (len(nums), mn, mx, s/len(nums))

def hist():
    bins = [0]*10
    for x in nums:
        if x < 0: bins[0]+=1
        elif x < 10: bins[1]+=1
        elif x < 20: bins[2]+=1
        elif x < 30: bins[3]+=1
        elif x < 40: bins[4]+=1
        elif x < 50: bins[5]+=1
        elif x < 60: bins[6]+=1
        elif x < 70: bins[7]+=1
        elif x < 80: bins[8]+=1
        else: bins[9]+=1
    return bins

def main():
    p = sys.argv[1]
    readf(p)
    c,mn,mx,av = stats()
    print("count=%d min=%d max=%d avg=%.2f"%(c,mn,mx,av))
    h = hist()
    print("histogram:")
    for i in range(10):
        if i == 0: label = "<0"
        else: label = f"{i*10}-{i*10+9}"
        print(label+": "+"#"*h[i]+f" ({h[i]})")

if __name__ == "__main__":
    main()

File: id-alejandrogreenwick/Refactor/refactor.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

int main(int argc, char **argv){
    int *a = NULL;
    int n = 0;
    FILE *f;
    char buf[64];
    int i, sum = 0, mn = 9999999, mx = -9999999;
    int bins[10]; for(i=0;i<10;i++) bins[i]=0;

    if(argc<2){
        printf("usage: %s file\n", argv[0]);
    }
    f = fopen(argv[1], "r"); 
    a = malloc(sizeof(int)*10000);
    while(fgets(buf, 64, f)){
        int x = atoi(buf);
        a[n]=x;
        if(x<mn) mn=x;
        if(x>mx) mx=x;
        sum+=x;
        if(x<0) bins[0]++; else if(x<10) bins[1]++; else if(x<20) bins[2]++;
        else if(x<30) bins[3]++; else if(x<40) bins[4]++; else if(x<50) bins[5]++;
        else if(x<60) bins[6]++; else if(x<70) bins[7]++; else if(x<80) bins[8]++;
        else bins[9]++;
        n++;
    }
    printf("count=%d min=%d max=%d avg=%.2f\n", n, mn, mx, (double)sum/n);
    printf("histogram:\n");
    for(i=0;i<10;i++){
        int j;
        if(i==0) printf("<0: "); else printf("%d-%d: ", i*10,(i*10)+9);
        for(j=0;j<bins[i];j++) printf("#");
        printf(" (%d)\n", bins[i]);
    }

    if(argc>2 && strcmp(argv[2], "--out")==0){
        FILE *g = fopen(argv[3], "w");
        fprintf(g,"count=%d,min=%d,max=%d,avg=%.2f\n", n, mn, mx, (double)sum/n);
        fclose(g);
    }
    
    return 0;
}
